### PR TITLE
[test] CI should fail for new warnings

### PR DESCRIPTION
**Pull request by oliviertassinari**
This objective of this pull request is to surface an issue we will face with https://github.com/mui-org/material-ui/issues/19706. The equivalent behavior can be found in https://github.com/mui-org/material-ui/blob/14834adf8c07cd043a169cb486097f2363bbe420/test/utils/consoleError.js.

Waiting for #1776 to be resolved.

### CODE REVIEW

File: lib/src/__tests__/setup.js [hunk 1]
```diff
@@ -36,3 +36,24 @@ console.error = (...args) => {
     throw new Error(message);
   }
 };
+
+// Convert any console warning into a thrown error
+const warn = console.warn;
+console.warn = (...args) => {
+  warn.apply(console, args);
+  if (args[0] instanceof Error) {
```
Comment (dmtrKovalenko) [inline lib/src/__tests__/setup.js:44]: I think it will not work for warn, right?

File: lib/src/__tests__/setup.js [hunk 2]
```diff
@@ -36,3 +36,24 @@ console.error = (...args) => {
     throw new Error(message);
   }
 };
+
+// Convert any console warning into a thrown error
+const warn = console.warn;
+console.warn = (...args) => {
+  warn.apply(console, args);
```
Comment (dmtrKovalenko) [inline lib/src/__tests__/setup.js:43]: Probably we can move this logic up to the reusable function, and use it for both console.error and console.warn

**Review by dmtrKovalenko — CHANGES_REQUESTED** [review]

File: lib/src/__tests__/setup.js (hunk 1 above)
Comment (oliviertassinari) [reply lib/src/__tests__/setup.js:44]: I don't know. In the version of the module in the main repository, we assume the arguments is an array of strings.

```jsx
  console.error = (...args) => {
    // Can't use log as karma is not displaying them.
    console.info(...args);
    throw new Error(...args);
  };

  console.warn = (...args) => {
    // Can't use log as karma is not displaying them.
    console.info(...args);
    throw new Error(...args);
  };
```

We never had any issue with it yet. But it seems that the console methods can accept any object as input. What do you want to do?

File: lib/src/__tests__/setup.js (hunk 2 above)
Comment (oliviertassinari) [reply lib/src/__tests__/setup.js:43]: As you see fit, the module has a limited life expectancy.

**Review by oliviertassinari — COMMENTED** [review]
Thanks for the review. I would propose we wait for the resolution of #1776 before rebasing this one and merging.

**Pull request closed without merging**